examples/customresources/demos/complexmodule/module.go

```diff
@@ -21,7 +21,15 @@ import (
 )
 
 func main() {
-	utils.ContextualMain(mainWithArgs, golog.NewDevelopmentLogger("ComplexModule"))
+	// NewLoggerFromArgs can be used to create a golog.Logger at either
+	// "DebugLevel" or "InfoLevel" if -debug is or is not provided in os.Args
+	// respectively. It will ignore all other provided arguments.
+	logger, err := module.NewLoggerFromArgs("ComplexModule")
+	if err != nil {
+		panic(err)
```

Comment:
Generally we should never panic (it was explicitly called out in the anti-patterns talk Tess gave recently) and especially not at this high of a level. Just print the error and return.

Reply:
Thinking more on this, it's generally an anti-pattern in Go for a log constructor to be able to return an error (since logging is so critical to set up early.) I'll suggest changes to the constructor itself to simplify.